source/common/secret/secret_manager_impl.cc

```diff
@@ -0,0 +1,20 @@
+#include "common/secret/secret_manager_impl.h"
+
+#include "common/common/logger.h"
+#include "common/secret/secret_impl.h"
+
+namespace Envoy {
+namespace Secret {
+
+bool SecretManagerImpl::addOrUpdateStaticSecret(const SecretSharedPtr secret) {
+  static_secrets_[secret->name()] = secret;
+  return true;
```

Comment:
looks like it can't fail. Please make this a void function for now.

Reply:
Changed to void

Reply:
You can avoid a lot of those changes if you pass `SecretManager` to `Upstream::ProdClusterManagerFactory()` instead of passing it as the argument in all the calls.

Also, instead of passing and storing each of `TransportSocketFactoryContext`'s members (`Ssl::ContextManager`, `Secret::SecretManager` and `Stats::Store`) separately, we should be passing `TransportSocketFactoryContext` itself.

Reply:
(sorry, commented in wrong place)